src/dataServices/dynamoDbBundleService.ts

```diff
@@ -190,7 +190,8 @@ export class DynamoDbBundleService implements Bundle {
         const idItemsFailedToRead: string[] = [];
         for (let i = 0; i < itemResponses.length; i += 1) {
             const itemResponse = itemResponses[i];
-            if (itemResponse instanceof ResourceNotFoundError) {
+            // allow for update as create scenario
+            if (itemResponse instanceof ResourceNotFoundError && itemsToLock[i].operation !== 'update') {
```

Comment:
I think that this works, but it will inadvertently enable update as crate for all customers using bundles/transactions.

`DynamoDbDataService` has a parameter `supportUpdateCreate` that enables update as create for the regular update operation but doesn't address bundles/transactions. I think that we can pass down that same parameter to `DynamoDbBundleService` and use it to conditionally apply your changes.

https://github.com/awslabs/fhir-works-on-aws-persistence-ddb/blob/e913c711c842d922a9aa1902b6705d240af6ad68/src/dataServices/dynamoDbDataService.ts#L71

Reply:
I just added the `supportUpdateCreate` flag to dynamoDbBundleService.ts. Creating a test seems not so trivial. I'll give it a shot.